### ISSUE TITLE
Output includes a lot of unnecessary escape sequences

### ISSUE THREAD

**Issue by krader1961** [charmbracelet/glow#356]
I was working on https://github.com/elves/elvish/issues/1432 to implement a `help` command for the Elvish shell. The solution requires embedding documentation pre-rendered for a terminal. I really wanted to use `glow` to convert the Elvish Markdown documentation to literal text that looks good when written to a terminal. One small subset of the Elvish documentation results in 2.2 MB of output from `glow`. Using the Pandoc [terminal-writer filter](https://github.com/Orange-OpenSource/pandoc-terminal-writer) on the same input results in just 109 KB of output. The `glow` output is 20 times larger and very inefficient in terms of irrelevant, unnecessary, escape sequences.

Is this expected? I'm simply piping some Markdown into glow like this: `... | glow -s light -w 80 - | ...`. Replacing `glow` with `pandoc --from=gfm+attributes --to=terminal.lua` produces less aesthetically pleasing output but output that is 20x smaller. Is there some way to elide the redundant escape sequences?

**Comment by krader1961** [comment]
Also, a lot of non-Markdown syntax results in unexpected ANSI X3.64 markup. For example, in my use case there are lines such as `!!elvdoc:var:builtin:before-chdir`. Those lines are not embedded in any context that would justify adding ANSI X3.64 markup. Yet `glow` does so and also inserts such markup between the two exclamation marks. Why?

**Comment by muesli** [comment]
glow/glamour's output blocks are intentionally self-contained, the renderer does not keep a state of current output settings. You can compress the output to get rid of unnecessary escape sequences. See: https://github.com/muesli/ansi

Got any examples for X.364 markups? OSCs are only used on startup to determine your terminal's background color. You can change that behavior using flags.

**Comment by krader1961** [comment]
@muesli, I tried your ANSI compression tool. It had almost zero effect on the `glow` output. In fact, it slightly increased the size of the output (`x` has the `glow` output and `y` has the `ansi` output):

```
-rw-r----- 1 krader wheel 2475095 Jul 11 20:38 /tmp/x
-rw-r----- 1 krader wheel 2476557 Jul 12 14:08 /tmp/y
```

The good news is I took another look at the `glow` documentation and realized using a custom style file, rather than the builtin `light` style, would solve the problem. Specifically, removing this line `"color": "234",` from the document definition. That eliminated all the unexpected ANSI sequences and resulted in output approximately the same size as produced by `pandoc`.
